PowerEditor/src/Notepad_plus.cpp

```diff
@@ -3103,14 +3103,25 @@ void Notepad_plus::maintainIndentation(TCHAR ch)
 	intptr_t tabWidth = _pEditView->execute(SCI_GETTABWIDTH);
 
 	LangType type = _pEditView->getCurrentBuffer()->getLangType();
+	LangAutoIndentType autoIndentType = LangAutoIndentType::Standard;
+
+	// For external languages, query for custom auto-indentation funcionality
+	if (type >= L_EXTERNAL)
+	{
+		NppParameters& nppParam = NppParameters::getInstance();
+		autoIndentType = nppParam.getELCFromIndex(type - L_EXTERNAL)._autoIndentType;
+		if (autoIndentType == LangAutoIndentType::Custom)
+			return;
+	}
 
 	// Do not alter indentation if we were at the beginning of the line and we pressed Enter
 	if ((((eolMode == SC_EOL_CRLF || eolMode == SC_EOL_LF) && ch == '\n') ||
 		(eolMode == SC_EOL_CR && ch == '\r')) && prevLine >= 0 && _pEditView->getLineLength(prevLine) == 0)
 		return;
 
 	if (type == L_C || type == L_CPP || type == L_JAVA || type == L_CS || type == L_OBJC ||
-		type == L_PHP || type == L_JS || type == L_JAVASCRIPT || type == L_JSP || type == L_CSS || type == L_PERL || type == L_RUST || type == L_POWERSHELL || type == L_JSON)
+		type == L_PHP || type == L_JS || type == L_JAVASCRIPT || type == L_JSP || type == L_CSS || type == L_PERL || 
+		type == L_RUST || type == L_POWERSHELL || type == L_JSON || autoIndentType == LangAutoIndentType::Extended)

```

Comment:
If my description above about `Custom` mode is correct, then why is `autoIndentType == LangAutoIndentType::Extended` needed?

Reply:
It is, because Extended will allow notepad to perform a C-Style indentation. My language is a C-Derived one, so the same rules apply. Hence, with an "extended" I don't need to rewrite code (nor any future C-Like plugin).